kv/index.go

```diff
@@ -195,16 +195,18 @@ func (i *Index) Walk(ctx context.Context, tx Tx, foreignKey []byte, visitFn Visi
 		return err
 	}
 
-	return indexWalk(ctx, cursor, sourceBucket, visitFn)
+	return indexWalk(ctx, foreignKey, cursor, sourceBucket, visitFn)
 }
 
 // indexWalk consumes the indexKey and primaryKey pairs in the index bucket and looks up their
 // associated primaryKey's value in the provided source bucket.
 // When an item is located in the source, the provided visit function is called with primary key and associated value.
-func indexWalk(ctx context.Context, indexCursor ForwardCursor, sourceBucket Bucket, visit VisitFunc) (err error) {
+func indexWalk(ctx context.Context, foreignKey []byte, indexCursor ForwardCursor, sourceBucket Bucket, visit VisitFunc) (err error) {
 	var keys [][]byte
 	for ik, pk := indexCursor.Next(); ik != nil; ik, pk = indexCursor.Next() {
-		keys = append(keys, pk)
+		if string(foreignKey) == string(ik) {
```

Comment:
```suggestion
		if bytes.Equal(foreignKey, bytes.SplitN(ik, []byte{'/'}, 2)[0]) {
```
I get it now :D This might be what you're looking for. The foreign key is all the bits before the `/` in the index key.

Reply:
or even:
```go
if ok := bytes.HasPrefix(ik, foreignKey); ok && len(ik) > len(foreignKey) && ik[len(foreignKey)] == '/' {

}
```

Reply:
Thanks for the info – I see now that the `Index` composes the key as `foreignKey` + `/` + `primaryKey`

Reply:
I have resolved the issue with the latest changes by using the `indexKeyParts` function to parse the foreign key value. Thanks again for leading me in the right direction 👏